data_structures/linked_list/floyds_cycle_detection.py

```diff
@@ -0,0 +1,136 @@
+"""
+Floyd's cycle detection algorithm is a popular algorithm used to detect cycles
+in a linked list. It uses two pointers, a slow pointer and a fast pointer,
+to traverse the linked list. The slow pointer moves one node at a time while the fast
+pointer moves two nodes at a time. If there is a cycle in the linked list,
+the fast pointer will eventually catch up to the slow pointer and they will
+meet at the same node. If there is no cycle, the fast pointer will reach the end of
+the linked list and the algorithm will terminate.
+
+For more information: https://en.wikipedia.org/wiki/Cycle_detection#Floyd's_Tortoise_and_Hare
+"""
+
+from collections.abc import Iterator
+from dataclasses import dataclass
+from typing import Any
+
+
+@dataclass
+class Node:
+    """
+    A class representing a node in a singly linked list.
+    """
+
+    data: Any
+    next_node: "Node" | None = None
```

Comment:
```suggestion
    next_node: Self | None = None
```
This line is causing the build to fail. As of Python 3.11, you can use the `Self` type to annotate a variable with the name of the enclosing class. I believe this should fix the build error.

Reply:
Oh! Let me try again!
Actually I tried to run the tests on my system. But I'm getting some errors - I couldn't fix any of those for 4 hrs. I'm getting one error if I use power shell and other error if I use WSL.

Reply:
@tianyizheng02 I delete the repo, I'll try again!
But I am still not able to understand why am I getting errors on my system, I also posted the same on discord.